src/components/details-card/details-card.tsx

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright contributors to the ibm-storage-odf-console project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Link, BrowserRouter as Router, } from 'react-router-dom';
+import { Base64 } from 'js-base64';
+import {
+    useK8sWatchResource,
+} from "@console/dynamic-plugin-sdk/api";
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+  DetailItem,
+  DetailsBody,
+} from '@console/dynamic-plugin-sdk/internalAPI';
+import {ExternalLink} from './Link';
+import { 
+  StorageInstanceKind, 
+  K8sKind,
+  SecretKind,
+ } from '../../types';
+import { 
+  getEndpoint,  
+  getIBMStorageODFVersion,
+  resourcePathFromModel,
+  } from '../../selectors';
+import {
+  ClusterServiceVersionModel,
+} from '../../models';
+import {
+  GetFlashSystemResource, 
+  SubscriptionResource,
+  GetSecretResource 
+} from '../../constants/resources'
+
+const DetailsCard: React.FC<any> = (props) => {
+  const flashClusterResource = GetFlashSystemResource(props?.match?.params?.name, props?.match?.params?.namespace);
+  const [data, flashsystemloaded, flashsystemloadError] = useK8sWatchResource<StorageInstanceKind>(flashClusterResource);
+  const [subscriptions, subscriptionloaded, subscriptionloadError] = useK8sWatchResource<K8sKind[]>(SubscriptionResource);
+
+  const stoData = data?.[0];
+  const flashSecretResource = GetSecretResource(stoData?.metadata?.name, stoData?.metadata?.namespace);
```

Comment:
is this namespace is same as props?.match?.params?.namespace?

if yes, then please move all hooks at the top, Also it is not safe to getting namespace from data and then use it here, if is changed or removed I thining the watcher might throw an error like it is not the same as the previous render.
```
const {namespace} = props?.match?.params;
const flashClusterResource = GetFlashSystemResource(props?.match?.params?.name, namespace);
const flashSecretResource = GetSecretResource(namespace);

// Hooks
const [data, flashsystemloaded, flashsystemloadError] = useK8sWatchResource<StorageInstanceKind>(flashClusterResource);

const [subscriptions, subscriptionloaded, subscriptionloadError] = useK8sWatchResource<K8sKind[]>(SubscriptionResource);
const [secret, secretloaded, secretloadError] = useK8sWatchResource<SecretKind>(flashSecretResource);

```

Reply:
if the namespace is different,  if it keep changes i am doubting watcher will break

Reply:
For a resource, namespace and name are key identifying parameters, IMO it wont be any issue.

Reply:
ack